### ISSUE TITLE
Replace `readme_file.get_file_contents` template with new generic `readme.md`

### ISSUE THREAD

**Issue by nozomione** [AlexsLemonade/scpca-portal#795]
### Context

In #794, we've changed the `templates/readme/` folder structure to allow a single `readme.md` template to generate a downloadable `README.md` file for all computed files. The `readme.md` template then includes the readme file in the "Contents" section using the template context `contents_template`, otherwise explicitly includes the readme files for the remaining sections.

We'll now need to dynamically map  this `contents_template` and provide it to the `readme.me` template. 

### Problem or idea

To accomplish this, we should modify `readme_file.get_file_contents` to do the following:
- Based on the given `download_config`, it should determine the corresponding readme file location of `contents` subfolder and store in `contents_template`
- Provide the following template contexts to `readme.md`:
  -  `contents_template` for the contents section
  - `download_config` to conditionally render the usage section based on `format`, `modality`, and `metadata_only`

### Solution or next step

Modify `readme_file.get_file_contents` and provide `contents_template` to the `readme.md` template:

e.g.) 
    
  ```python
  def get_file_contents(project: Project, download_config: Dict) -> str
      template_dir = common.TEMPLATE_PATH / "readme"
  
      # Find the contents template based on the download config
      readme_template_keys = [
          download_config["modality"],
          download_config["format"]
      ]
      
      if download_config["includes_merged"]:
          readme_template_keys.append("MERGED")
      elif download_config["includes_multiplexed"]:
          rreadme_template_keys.append("MULTIPLEXED")
      elif download_config["metadata_only"]
          readme_template_keys= ["METADATA_ONLY"]
     
      # Map the 'contents' readme file 
      contents_template = f"readme/contents/{'_'.join(readme_template_keys)}.md"
 
      # Provide contents_template to readme template
      return render_to_string(
          common.TEMPLATE_PATH / "readme" / "readme.md",
          context={
              "date": utils.get_today_string(),
              "contents_template": contents_template, # here
              "projects": [self],     
          }
      ).strip()
  ```

**Comment by davidsmejia** [comment]
You shouldn't need to create a readme sort order. The generic file will be a template that includes the nested templates. We only want to specify the "contents" section because those are all different depending on the `download_config`.

**Comment by nozomione** [comment]
> You shouldn't need to create a readme sort order. The generic file will be a template that includes the nested templates. We only want to specify the "contents" section because those are all different depending on the `download_config`.

I've adjusted the steps in #794 and in this issue. 

The common readme sort order is used to render the subsections in `README.md` in the correct order.

The entire flow would be:
1. In the `readme.md` template, we use `content_templates` to include the subfolder's file path using a `for` loop:
```markdown
 <!-- readme.md-->
 {% for content_template in content_templates %}
      {% for content, content_path in content_template.items %}
          {% include content_path %}
      {% endfor %}
  {% endfor %}
```
2. In `Project::create_readme_file`, we dynamically build `content_templates` based on `download_config:

e.g.) For the following `download_config`:
```py
    download_config = {
         "modality": "SINGLE_CELL",
         "format": "SINGLE_CELL_EXPERIMENT",
         "includes_multiplexed": False,
         "includes_merged": False,
         "metadata_only": False,
     }
```
`content_templates` should be:
```py
# a list of dictionaries containing all sections' file path (this is after sorted)
content_templates = [
      {"contents": "readme/contents/SINGLE_CELL_SINGLE_CELL_EXPERIMENT.md"},
      {"usage": "readme/usage/SINGLE_CELL_SINGLE_CELL_EXPERIMENT.md"},
      {"changelog": "readme/changelog/index.md"} ,
      {"contact": "readme/contact/index.md"}, 
      {"citation": "readme/citation/index.md"}, 
      {"terms_of_use": "readme/terms_of_use/index.md"},
      {"additional_terms": "readme/additional_terms/index.md"},
    ]
```

Alternatively, we may change the `content_templates` data type to a dictionary (instead of a list of dictionaries), then we can include files explicitly with the section names in the `readme.md` template. This approach eliminates the need for the common readme sort order, but requires editing the template every time the order of the sections changes etc.

```pt
content_templates = {
    "contents": "readme/contents/SINGLE_CELL_SINGLE_CELL_EXPERIMENT.md",
    "usage": "readme/usage/SINGLE_CELL_SINGLE_CELL_EXPERIMENT.md",
    "changelog": "readme/changelog/index.md",
    "contact": "readme/contact/index.md",
    "citation": "readme/citation/index.md",
    "terms_of_use": "readme/terms_of_use/index.md",
    "additional_terms": "readme/additional_terms/index.md",
}
```

```markdown
<!-- readme.md -->
{% include content_templates.contents %}
{% include content_templates.usage %}
{% include content_templates.changelog %}
{% include content_templates.contact %}
{% include content_templates.citation %}
{% include content_templates.terms_of_use %}
{% include content_templates.additional_terms %}
```
Please let me know your thought on this, David!

**Comment by avrohomgottlieb** [comment]
I personally like the idea of explicit inclusions and usage of a dictionary here. It feels like it'll be easier to debug if there's an issue with one of the readme components.

**Comment by nozomione** [comment]
> I personally like the idea of explicit inclusions and usage of a dictionary here. It feels like it'll be easier to debug if there's an issue with one of the readme components.

It makes perfect sense to keep things simple, especially for debugging! Thank you for your feedback, Avrohom!

**Comment by nozomione** [comment]
This issue and #794 have been updated. More adjustments to this issue may be done once the PR for #766 lands.

**Comment by nozomione** [comment]
Closed with #806 (this PR was also linked via Development of the right column).
